roomserver/storage/postgres/events_table.go

```diff
@@ -460,8 +465,13 @@ func (s *eventStatements) BulkSelectEventID(ctx context.Context, txn *sql.Tx, ev
 
 // bulkSelectEventNIDs returns a map from string event ID to numeric event ID.
 // If an event ID is not in the database then it is omitted from the map.
-func (s *eventStatements) BulkSelectEventNID(ctx context.Context, txn *sql.Tx, eventIDs []string) (map[string]types.EventNID, error) {
-	stmt := sqlutil.TxStmt(txn, s.bulkSelectEventNIDStmt)
+func (s *eventStatements) BulkSelectEventNID(ctx context.Context, txn *sql.Tx, eventIDs []string, onlyUnsent bool) (map[string]types.EventNID, error) {
```

Comment:
Questioning whether it would just be better to make this private and have two functions: `BulkSelectEventNID` and `BulkSelectUnsentEventNID`. It would be clearer and not need a flag which is unintuitive. I don't mind having this boolean inside a private function call, so you don't need to duplicate code.